applications/GeoMechanicsApplication/tests/cpp_tests/test_Hencky_strain.cpp

```diff
@@ -0,0 +1,95 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Wijtze Pieter Kikstra
+//
+
+#include "testing/testing.h"
+#include "custom_utilities/math_utilities.hpp"
+#include "custom_utilities/stress_strain_utilities.hpp"
+
+using namespace Kratos;
+
+namespace Kratos::Testing
+{
+
+KRATOS_TEST_CASE_IN_SUITE(CheckHenckyStrainIncompressibleUniaxialElongation, KratosGeoMechanicsFastSuite)
+{
+    double stretch = 2.0;
+    Matrix Fmatrix = ZeroMatrix(3,3);
+    Fmatrix(0,0) = stretch;
+    Fmatrix(1,1) = 1.0/std::sqrt(stretch);
+    Fmatrix(2,2) = Fmatrix(1,1);
+
+    Vector Evector = StressStrainUtilities::CalculateHenckyStrain(Fmatrix, 6);
+    KRATOS_EXPECT_DOUBLE_EQ(0.5*std::log(stretch*stretch), Evector[0]);
+    KRATOS_EXPECT_DOUBLE_EQ(0.5*std::log(1./stretch), Evector[1]);
+    KRATOS_EXPECT_DOUBLE_EQ(0.5*std::log(1./stretch), Evector[2]);
+    KRATOS_EXPECT_DOUBLE_EQ(0., Evector[3]);
```

Comment:
Just for my information: I don't understand why Evector has 4 elements, while the Matrix is 3x3

Reply:
The strain vector for plane strain contains eps_xx, eps_yy, eps_zz=0, eps_xy, 
for solids the strain vector has eps_xx, eps_yy, eps_zz, eps_xy, eps_yz, eps_zx.
I added checks on eps_yz = eps_zx = 0 ( so now all 6 components )